src/lib/EmissaryLogic.sol

```diff
@@ -0,0 +1,113 @@
+    using EmissaryLib for address;
+
+    error InvalidEmissaryAssignment();
+
+    /**
+     * @notice Initiates the timelock process for changing an emissary
+     * @dev This function starts the timelock period that must pass before
+     *      a new emissary can be set. The timelock is specific to the caller (msg.sender).
+     *      After calling this function, the caller must wait for the timelock period to expire
+     *      before calling `assignEmissary`.
+     *
+     *      The timelock mechanism ensures that changes to emissaries are not immediate,
+     *      providing a security buffer to prevent malicious or accidental redelegation attempts.
+     *      This period allows time for any necessary reviews or interventions.
+     *
+     *      The function utilizes `toAllocatorIdIfRegistered` to validate and convert the allocator
+     *      address to its corresponding ID, ensuring that only registered allocators can proceed.
+     *
+     * @custom:emits EmissaryTimelockSet event through the library call, signaling the start of the timelock period
+     */
+    function _scheduleEmissaryAssignment(address sponsor, address allocator, ResetPeriod resetPeriod, Scope scope) internal returns (uint256 emissaryAssignmentAvailableAt) {
+        bytes12 lockTag = allocator.toAllocatorIdIfRegistered().toLockTag(scope, resetPeriod);
+        emissaryAssignmentAvailableAt = sponsor.scheduleEmissaryAssignment(lockTag);
+    }
+
+    /**
+     * @notice Sets a new emissary for the sponsor
+     * @dev This function can only be called after the timelock period has passed.
+     *      The timelock must be initiated by calling `scheduleEmissaryAssignment` first.
+     *      The emissary address can be set to the zero address to remove delegation,
+     *      effectively disabling the current emissary.
+     *
+     *      The emissary address must not be the address of the allocator to prevent
+     *      conflicts of interest and maintain the integrity of the delegation mechanism.
+     *
+     *      The function utilizes `toAllocatorIdIfRegistered` to validate and convert the allocator
+     *      address to its corresponding ID, ensuring that only registered allocators can proceed.
+     *
+     *      The `ResetPeriod` parameter specifies the reset behavior for the emissary assignment,
+     *      adding flexibility to the delegation process.
+     *
+     * @param emissary The address that will be authorized to sign on behalf of the sponsor.
+     *                  Set to address(0) to remove the current emissary.
+     * @custom:emits EmissarySet event through the library call, signaling the successful assignment of a new emissary
+     * @custom:throws If the timelock period has not passed or was not initiated, ensuring secure delegation practices
+     */
+    function _assignEmissary(address sponsor, address allocator, address emissary, bytes calldata proof, ResetPeriod resetPeriod, Scope scope) internal returns (bool) {
```

Comment:
we should also drop it here *unless* we want to support automatic assignment of scheduled + ready emissaries on verification